### PR TITLE
Added the debug(addressSanitizer)  version of docker image.

### PR DESCRIPTION
**Pull request by monadbobo**
Added the debug(addressSanitizer)  version of docker image.

### CODE REVIEW

**Comment by monadbobo** [comment]
depend #1861

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -44,3 +44,35 @@ jobs:
...
+          fetch-depth: 1
+      - name: Compute tag
+        id: tag
+        run: |
+          [[ "${{ github.event_name }}" == "push" ]] && \
+            echo "::set-output name=tag::debug-nightly"
+        shell: bash
+      - name: Build image
+        env:
+          IMAGE_NAME: ${{ secrets.DOCKER_USERNAME }}/nebula-${{ matrix.service }}:${{ steps.tag.outputs.tag }}
```
Comment (laura-ding) [inline .github/workflows/docker.yaml]: make `IMAGE_NAME ` different with `docker`?

Comment (monadbobo) [reply .github/workflows/docker.yaml]: yes.

**Comment by monadbobo** [comment]
depend #1941

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -44,3 +44,35 @@ jobs:
           docker build -t ${IMAGE_NAME} -f docker/Dockerfile.${{ matrix.service }} .
           docker push ${IMAGE_NAME}
         shell: bash
+  docker-debug:
+    name: build docker debug image
+    runs-on: ubuntu-latest
```
Comment (yixinglu) [inline .github/workflows/docker.yaml:49]: Should only allow `push` event to start this job by:

```yaml
if: github.event_name == "push"
```

File: .github/workflows/docker.yaml [hunk 3]
```diff
@@ -44,3 +44,35 @@ jobs:
...
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Compute tag
+        id: tag
+        run: |
+          [[ "${{ github.event_name }}" == "push" ]] && \
+            echo "::set-output name=tag::debug-nightly"
+        shell: bash
```
Comment (yixinglu) [inline .github/workflows/docker.yaml:71]: After above condition, this step could be removed.

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (yixinglu) [inline .github/workflows/docker.yaml:74]: Specify this docker image tag to `debug-nightly`

File: .github/workflows/docker.yaml [hunk 4]
```diff
@@ -44,3 +44,35 @@ jobs:
...
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Compute tag
+        id: tag
+        run: |
+          echo "::set-output name=tag::debug-nightly"
+        shell: bash
```
Comment (yixinglu) [inline .github/workflows/docker.yaml:71]: It seems that you can delete this step.

**Comment by CPWstatic** [comment]
I'll close it first since its not strongly demanded. Please pull another request to v2.0 if we really need it.

**Pull request closed without merging**